// @trapi/query/dist/constants.js
"use strict";
/*
 * Copyright (c) 2021-2022.
 * Author Peter Placzek (tada5hi)
 * For the full copyright and license information,
 * view the LICENSE file that was distributed with this source code.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.URLParameter = exports.Parameter = void 0;
// -----------------------------------------------------------
var Parameter;
(function (Parameter) {
    Parameter["FILTERS"] = "filters";
    Parameter["FIELDS"] = "fields";
    Parameter["PAGINATION"] = "pagination";
    Parameter["RELATIONS"] = "relations";
    Parameter["SORT"] = "sort";
})(Parameter = exports.Parameter || (exports.Parameter = {}));
// -----------------------------------------------------------
var URLParameter;
(function (URLParameter) {
    URLParameter["FILTERS"] = "filter";
    URLParameter["FIELDS"] = "fields";
    URLParameter["PAGINATION"] = "page";
    URLParameter["RELATIONS"] = "include";
    URLParameter["SORT"] = "sort";
})(URLParameter = exports.URLParameter || (exports.URLParameter = {}));
